### PR TITLE
checkMissingsByValues, partially concerns #4

### PR DESCRIPTION
**Pull request by beckerbenj**
This adresses a feature request from the FDZ, but could be also valuable for BT etc.

The PR also cleans up the tests and documentation of `checkMissings()`.

Please review till Friday.

### CODE REVIEW

**Comment by nickhaf** [comment]
Ich habe mich mal für die Pull Request selbst assigned, das können wir ja immer machen wenn wir ein Review starten. Also ich schaus mir jetzt mal an :D

File: R/checkMissings.R [hunk 1]
```diff
@@ -2,25 +2,37 @@
 #############################################################################
 #' Check and Adjust Missing Coding
 #'
-#' Function to check if missings are coded and labeled correctly in a \code{GADSdat} object.
+#' Functions to check if missings are coded and labeled correctly in a \code{GADSdat} object.
```
Comment (nickhaf) [inline R/checkMissings.R:5]: Nit: You could also use \`GADSdat\` to show code.

File: R/checkMissings.R [hunk 2]
```diff
@@ -2,25 +2,37 @@
...
 #'
 #'@param GADSdat \code{GADSdat} object imported via \code{eatGADS}.
-#'@param missingLabel Single string indicating how missing labels are commonly named in the value labels.
-#'@param addMissingCode If \code{TRUE}, missing codes are added according to occurrence of \code{"missingLabel"} in \code{"valLabel"}.
+#'@param missingLabel Single regular expression indicating how missing labels are commonly named in the value labels.
+#'@param missingValues Numeric vector of values which are commonly used for missing values.
+#'@param addMissingCode If \code{TRUE}, missing codes are added according to  \code{missingLabel} or \code{missingValues}.
 #'@param addMissingLabel If \code{TRUE}, \code{"generic missing"} is added according to occurrence of \code{"miss"} in \code{"missings"}. As often various value labels for missings are used, this argument should be used with great care.
 #'
 #'@return Returns a \code{GADSdat} object.
```
Comment (nickhaf) [inline R/checkMissings.R:20]: Nit: Maybe make more clear what is changed in the new `GADSdat` object

File: R/checkMissings.R [hunk 3]
```diff
@@ -57,6 +70,36 @@ insert_string <- function(df, rows, col, string) {
...
+checkMissingsByValues.GADSdat <- function(GADSdat, missingValues = -50:-99, addMissingCode = TRUE) {
+  check_GADSdat(GADSdat)
+  labels <- GADSdat$labels
+
+  missCode_missing_rows <- which(labels$value %in% missingValues & (is.na(labels$missings) | labels$missings == "valid"))
+  missCode_toomuch_rows <- which(labels$missings == "miss" & labels$value %in% missingValues)
+
+  ## Which variables are affected, how many adjustments are performed
+  if(length(missCode_missing_rows) > 0) {
+    message("The following variables have values in the 'missingValues' range which are not coded as missing:\n",
```
Comment (nickhaf) [inline R/checkMissings.R:89]: Nit: Maybe add an information that the missings are updated, if the argument was set to `TRUE`?

**Review by nickhaf — APPROVED** [review]
Hat super funktioniert, konnte den branch auch pullen und direkt alles ausprobieren.

File: R/checkMissings.R (hunk 1 above)
Comment (beckerbenj) [reply R/checkMissings.R:5]: was not aware of that feature, nice!

Comment (beckerbenj) [reply R/checkMissings.R:5]: (will not be implemented due to incompatabilities):
[roxygen2](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd-formatting.html)

**Pull request merged**